extensions/wikia/PortableInfobox/services/PortableInfoboxDataService.class.php

```diff
@@ -233,7 +233,21 @@ protected function clear() {
 		if ( $id ) {
 			$this->propsProxy->set( $id, [ self::INFOBOXES_PROPERTY_NAME => '' ] );
 			// don't cache clear state
-			$this->purge();
+			$this->invalidateCache();
 		}
 	}
+
+	/**
+	 * This method performs rendering of a current template and saves its metadata
+	 * in the page_props table.
+	 *
+	 * This method will be used to fix portable templates with missing database entries.
+	 *
+	 * @see CORE-6
+	 */
+	public function regenerate() {
```

Comment:
where is this being used?

Reply:
It will be used by a following script that fixes the issue for affected portable infoboxes:

```php
$pages = wfGetDB(DB_SLAVE)->selectFieldValues('page', 'page_id', 'page_namespace = ' . NS_TEMPLATE);
foreach($pages as $page) PortableInfoboxDataService::newFromPageID($page)->regenerate();
```